sympy/calculus/util.py

```diff
@@ -343,6 +343,17 @@ def periodicity(f, symbol, check=False):
             period = f.period(symbol)
         except NotImplementedError:
             pass
+    if isinstance(f,Abs):
+        f=f.args[0]
+        if isinstance(f,sec):
```

Comment:
I don't follow why you are recasting the function to cos or sin. Aren't these functions already handled by periodicity? It *seems* like you would just do:

```
period = periodicity(expr, symbol)
if period and isinstance(expr, TrigonometricFunction):
    period /= 2
```

I believe this is even true for `tan(x)` -- although `abs(tan(x))` will have a slope discontinuity, the function itself has a periodicity of pi.

I can't test this on sympy.live: what is `periodicity(cos(x) + S.Half, x)`? It should be `2*pi`

Reply:
We have to check whether **period/2** is really a period of a function like **Abs(f(x))**. for that f(x+period/2) have to be either f(x) or -f(x)  
```
if f.subs(symbol, symbol + period/2) in [f,-f]:
 ```
without this **if f.subs(symbol, symbol + period/2) in [f,-f]:** condition ```periodicity(Abs(sin(sin(x)+tan(x))), x)``` will return half of ```periodicity(sin(sin(x)+tan(x)),x)``` which is **pi** and **pi** is not a period of (Abs(sin(sin(x)+tan(x)))

I think that The period of abs(sec(x)) is **pi**  but for sympy sec(x+pi)=-1/cos(x) which is not same as -sec(x)
```
>>> from sympy import *
>>> x=Symbol('x')
>>> sec(x+pi)
-1/cos(x)
>>> sec(x+pi)==(-sec(x))
False
```
but for sec(x) or csc(x)  that if condition will always return false.So periodicity will return period instead of period/2.That's why I want to change sec(x) to cos(x).

Reply:
periodicity(cos(x) + S.Half, x) is 2*pi